lib/screens/authentication/recover/recover_account_search/recover_account_screen.dart

```diff
@@ -104,20 +109,9 @@ class _RecoverAccountScreenState extends State<RecoverAccountScreen> {
       builder: (context) {
         return RecoverAccountConfirmationDialog(
           account: lostAccount,
-          onConfirm: () async {
+          onConfirm: () {
+            BlocProvider.of<RecoverAccountSearchBloc>(buildContext).add(const OnConfirmRecoverTapped());
             Navigator.pop(context);
-            // TODO(n13): This needs to actually initialize the recovery!
-
-            // TODO here:
-            // maybe pop the dialog, then show a progress indicator (this takes 6 seconds! an eternity!)
-            // then show the next screen, recoverAccount details
-            // or it could show a "recovery initiated successfully" dialog, then on OK on that move on to Routes.recoverAccountDetails
-            // anyway it needs to end up in Routes.recoverAccountDetails so the user can share the link etc (that part already works)
-            final address = accountService.currentAccount.address;
-            await polkadotRepository.recoveryRepository.initiateRecovery(rescuer: address, lostAccount: lostAccount);
```

Comment:
making network alls from the UI? Shame! 
I removed this from the UI and moved to the bloc. 

The bloc also handles loading, success and error message now.

Reply:
<img width="434" alt="image" src="https://user-images.githubusercontent.com/8041901/193463104-a4c9bb52-ed75-4eb2-910c-cea4d49927cf.png">

Reply:
Yep that was the idea thank you! :)